test/mockupdb/test_handshake.py

```diff
@@ -11,14 +11,56 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-
-from mockupdb import MockupDB, OpReply, OpMsg, absent, Command, go
+import mockupdb
+from mockupdb import (MockupDB, OpReply, OpMsg, OpMsgReply, OpQuery, absent,
+                      Command, go, Request)
 from pymongo import MongoClient, version as pymongo_version
 from pymongo.errors import OperationFailure
+from pymongo.server_api import ServerApi
 
 import unittest
-
+from copy import deepcopy
+import time
+
+def test_handshake_with_option(self, protocol, num_servers, **kwargs):
+    primary = MockupDB(verbose=False)
+    secondaries = [MockupDB(verbose=False) for _ in range(num_servers-1)]
+    servers = [primary]+secondaries
+    hello = "ismaster" if isinstance(protocol(), OpQuery) else "hello"
+
+    # set up a custom handler to log and then immediately close all connections
+    # this makes sure that we only are checking the hello command, because
+    # after the first response it has been closed
+    reqs = []
+    def hangup(response):
+        reqs.append(deepcopy([response.doc, response.__class__]))
```

Comment:
Why is the deepcopy needed? Can we do `reqs.append(request)`?

Reply:
This was I think due to having the same autoresponder on every server. It is fixed now.